### ISSUE TITLE
"useradd: group foo exists - if you want to add this user..."

### ISSUE THREAD

**Issue by drewp** [Fizzadar/pyinfra#835]
## Describe the bug

Possibly related to #700.

My code:
```
    server.group(group='elastic', gid=3018)
    server.user(user='elastic', uid=3018, group='elastic')

    server.group(group='drewnote', gid=1009)
    server.user(user='drewnote', uid=1009)
```
Target host says this:
```
% ls /home/{drewnote,elastic}
ls: cannot access '/home/drewnote': No such file or directory
ls: cannot access '/home/elastic': No such file or directory

drewp@pipe ~ % egrep '^elastic|^drew' /etc/passwd /etc/group 
/etc/passwd:drewp:x:1001:501::/home/drewp:/bin/sh
/etc/group:drewp:x:501:
```
(i.e. drewp exists but not drewnote) and my run goes like this:
```

--> Loading config...
--> Loading inventory...

--> Connecting to hosts...
    [pipe] Connected

--> Preparing operations...
    Loading: users.py
[pipe] >>> sh -c 'cd /etc/ && for file in $(ls -pdL *-release | grep -v /); do echo "/etc/${file}"; cat "/etc/${file}"; echo ---; done'
    [pipe] Loaded fact server.LinuxDistribution
[pipe] >>> sh -c 'cat /etc/group'
    [pipe] Loaded fact server.Groups
[pipe] >>> sh -c 'for i in `cat /etc/passwd | cut -d: -f1`; do
            ENTRY=`grep ^$i: /etc/passwd`;
            LASTLOG=`lastlog -u $i | grep ^$i` | tr -s '"'"' '"'"';
            echo "$ENTRY|`id -gn $i`|`id -Gn $i`|$LASTLOG";
        done'
    [pipe] Loaded fact server.Users
[pipe] >>> sh -c '! (test -e /home/elastic || test -L /home/elastic ) || ( stat -c '"'"'user=%U group=%G mode=%A atime=%X mtime=%Y ctime=%Z size=%s %N'"'"' /home/elastic 2> /dev/null || stat -f '"'"'user=%Su group=%Sg mode=%Sp atime=%a mtime=%m ctime=%c size=%z %N%SY'"'"' /home/elastic )'
    [pipe] Loaded fact files.Directory (path=/home/elastic)
[pipe] >>> sh -c '! (test -e /home/drewnote || test -L /home/drewnote ) || ( stat -c '"'"'user=%U group=%G mode=%A atime=%X mtime=%Y ctime=%Z size=%s %N'"'"' /home/drewnote 2> /dev/null || stat -f '"'"'user=%Su group=%Sg mode=%Sp atime=%a mtime=%m ctime=%c size=%z %N%SY'"'"' /home/drewnote )'
    [pipe] Loaded fact files.Directory (path=/home/drewnote)
    [pipe] Ready: users.py

--> Proposed changes:
    Groups: small / inventory
    [pipe]   Operations: 4   Commands: 8   

--> Beginning operation run...
--> Starting operation: Server/Group (group=elastic, gid=3018)
[pipe] >>> sh -c 'grep '"'"'^elastic:'"'"' /etc/group || groupadd elastic --gid 3018'
    [pipe] Success

--> Starting operation: Server/User (user=elastic, uid=3018, group=elastic)
[pipe] >>> sh -c 'grep '"'"'^elastic:'"'"' /etc/passwd || useradd -d /home/elastic -g elastic --uid 3018 elastic'
[pipe] >>> sh -c 'mkdir -p /home/elastic'
[pipe] >>> sh -c 'chown elastic:elastic /home/elastic'
    [pipe] Success

--> Starting operation: Server/Group (group=drewnote, gid=1009)
[pipe] >>> sh -c 'grep '"'"'^drewnote:'"'"' /etc/group || groupadd drewnote --gid 1009'
    [pipe] Success

--> Starting operation: Server/User (user=drewnote, uid=1009)
[pipe] >>> sh -c 'grep '"'"'^drewnote:'"'"' /etc/passwd || useradd -d /home/drewnote --uid 1009 drewnote'
    [pipe] useradd: group drewnote exists - if you want to add this user to that group, use -g.
    [pipe] Error
--> pyinfra error: No hosts remaining!
```

## Meta

🛁 bang(pts/3):/my/proj/infra% env/bin/pyinfra --support

    System: Linux
      Platform: Linux-5.13.0-44-generic-x86_64-with-glibc2.34
      Release: 5.13.0-44-generic
      Machine: x86_64
    pyinfra: v2.2
    Executable: env/bin/pyinfra
    Python: 3.9.9 (CPython, GCC 10.3.0)

**Comment by Fizzadar** [comment]
So it looks like by default `user add` will create a group with the same name as the user which is causing this clash (https://stackoverflow.com/questions/28734496/linux-useradd-group-user1-exists-if-you-want-to-add-this-user-to-that-group).

One immediate workaround would be to just call `server.user` and have that create the user. Proper fix is adding a check if the group already exists and specifying it explicitly per he SO answer above.

**Comment by Fizzadar** [comment]
Added a fix for this case in https://github.com/Fizzadar/pyinfra/commit/5765007be766012a985310e25a227736f916341c, now released in v2.3 (and quick test just added in https://github.com/Fizzadar/pyinfra/commit/0a74ac342d96917e5c542e987ff006e26561de18).
